File: doc-examples/example-protobuf-java/src/main/java/example/ReadingCodec.java
package example;

import io.micronaut.serde.protobuf.ProtobufMapper;
import jakarta.inject.Singleton;

import java.io.IOException;

// tag::clazz[]
@Singleton
public class ReadingCodec {

    private final ProtobufMapper protobufMapper;   // <1>

    public ReadingCodec(ProtobufMapper protobufMapper) {
        this.protobufMapper = protobufMapper;
    }

    public byte[] encode(SensorReading reading) throws IOException {
        return protobufMapper.writeValueAsBytes(reading);
    }

    public SensorReading decode(byte[] payload) throws IOException {
        return protobufMapper.readValue(payload, SensorReading.class);
    }
}
// end::clazz[]

File: doc-examples/example-protobuf-kotlin/src/main/kotlin/example/ReadingCodec.kt
package example

import io.micronaut.serde.protobuf.ProtobufMapper
import jakarta.inject.Singleton

// tag::clazz[]
@Singleton
class ReadingCodec(
    private val protobufMapper: ProtobufMapper   // <1>
) {

    fun encode(reading: SensorReading): ByteArray =
        protobufMapper.writeValueAsBytes(reading)

    fun decode(payload: ByteArray): SensorReading =
        protobufMapper.readValue(payload, SensorReading::class.java)!!
}
// end::clazz[]
